_posts/2024-01-19-trino-2023-wrapped.md

```diff
@@ -0,0 +1,258 @@
+* [The main Trino Gateway repo](https://github.com/trinodb/trino-gateway/tree/main)
+
+## New features!
+
+### SQL routines (User-defined functions!)
+
+Whether you want to refer to them as SQL routines or as user-defined functions,
+they’re a big deal. Fresh off the presses and only a few months old, they do
+exactly what you’d expect them to do: you, a user, can define and re-use your
+own functions! Define and use them inline as part of a query to make that query
+cleaner, easier, and simpler to understand. Or, if you’re really cooking, you
+can run a query that defines the routine in the schema of the catalog. This
+allows other Trino users querying it to access the same function time and time
+again as part of their other queries. It’s a level of customization that we’ve
+never had before in Trino, and no longer do you need to write your own Java
+plugins in order to create and re-use functions that do exactly what you need
+them to do.
+
+If you want to learn more about SQL routines, you can check
+out [the introduction to SQL routines](/docs/current/routines/introduction.html)
+in our documentation, as well as a
+few [example routines](/docs/current/routines/examples.html) which give a good
+look at how they can be used.
+
+### Schema evolution and dynamic catalogs
+
+While we’re providing more power, customization, and flexibility to Trino users,
+it’s also important to highlight just how much has been added this year to make
+it easier to adjust things on the fly.
+
+Schema evolution in Hive came first, allowing you to alter columns’ data types,
+rename columns, and handle nested fields when dropping columns. Instead of
+needing to use the underlying database or modify it some other way and reboot
+Trino, Trino is now aware of when your schema changes and can handle the
+adjustments on the fly.
+
+But if you don't use Hive and are feeling left out, we’ve experimentally taken
+things one step further in 2023, adding dynamic catalogs to Trino. Rather than
+adjusting your schema one column at a time, what about adding or dropping an
+entire catalog in one go? You can do that now. Though it’s currently still
+bleeding-edge and not ready for widespread use on your important production
+data sources, we’re looking forward to improving it and making it resilient and
+stable in 2024.
+
+### Project Hummingbird
+
+Trino has always been about squeezing out every ounce of performance that you
+can get. You can go check out our [release notes](/current/release.html) and
+you’ll see that every version has at least a couple performance improvements in
+it, if not more. Over time, these performance improvements add up to a
+substantial gain, meaning that version-over-version, year-over-year, Trino is
+always getting faster. Project Hummingbird was a concerted effort this year to
+take a look at the core engine and make a number of small improvements that
+would add up to something
```

Comment:
It's more than "small improvements". Some are big architectural changes to how the core execution works.